src/S3ClientGUI/qtaskmodel.cpp

```diff
@@ -103,6 +111,25 @@ Qt::ItemFlags QTaskModel::flags(const QModelIndex &index) const
     return (QAbstractItemModel::flags(index)) & ~ Qt::ItemIsEditable;
 }
 
+void QTaskModel::reDownload(QModelIndex &index)
```

Comment:
No, you should create a "New task" , I do not like the idea of reusing the same s3 instance